packages/folio/init.lua

```diff
@@ -71,7 +79,16 @@ function package:registerCommands ()
   end, "Deprecated")
 
   self:registerCommand("foliostyle", function (_, content)
-    SILE.call("center", {}, content)
+    local style = SILE.settings:get("folio.style") 
+    if style == "mirror" then
```

Comment:
I like the feature and we should definitely support some way to handle this out of the box (since our currently documented "easy" way doesn't actually work). But this approach is just a bit too naive. In a RTL document where the writing direction is reversed, the "mirrored" pages are also the opposite of what most LTR readers expect: in a bound books odd pages are on the left and even pages are on the right.

Reply:
Would this solve the point?

```lua
      local ltr = (SILE.settings:get("font.direction") == "LTR")
      local evenPage = SILE.scratch.counters.folio.value % 2 == 0
        
      if (ltf and evenPage) or (not ltr and not evenPage) then
        SILE.call("raggedleft", {}, content)
      else
        SILE.call("raggedright", {}, content) 
      end
```

Reply:
Maybe. I haven't looked into when/where this is typically *set* vs. when/where this would be trying to read it. Also it would be complicated in a mixed-language environment. For example how would this folio function cope with a master frame layout for a parallel English/Arabic text with balanced frames? I'm wondering if a more robust setting might not be just to call out weather the "right" and "left" alignments should be on odd or even pages directly without trying to guess from the direction settings of the typesetter at some random point or some frame when there might be other considerations. Also what if people want to use the mirroring but just have a layout that is flipped from this expectation? Maybe they want their page numbers on the inside instead of the outside.

Thinking out loud here, but I'm leading towards a more explicit setting rather than trying to guess.

Reply:
It might be confusing too, if the folio frame doesn't span the whole page in some layouts (left/right mirrored alignment applying to the frame content regardless of its actual position) -- But I don't really know either, yet, what would be a proper way to better define it.

Reply:
In almost all cases that my pea brain on 2 hours of sleep can think off off the cuff, if the folio frame does not span the whole page then almost certainly *centered* is going to be the wrong expectation. In those cases being aligned to one side or the other seems like it would be expected.

This does bring up an interesting issue we have with odd/even headers as well. Mirroring the frame isn't enough to reflect the design, we also have style functions that vary for different sides (in the case of headers, sometimes also for content of course but that's different).

I'm even more convinced today than I was yesterday that we shouldn't be guessing at all and should just take some explicit option. But is it possible we have some property of the frame itself that could be used for a typesetter to derive the intentions while also allowing the masters package to mirror it?

Reply:
Of course this wouldn't work for documents with multiple columns, since it's just a draft... But that's one of the tests to be made as well, right?

```lua
 if SILE.frames.folio:right() == SILE.frames.content:right() and
    SILE.frames.folio:left() == SILE.frames.content:left() then
    ...
```